sqlmesh/core/macros.py

```diff
@@ -539,158 +539,30 @@ def filter_(evaluator: MacroEvaluator, *args: t.Any) -> t.List[t.Any]:
-        Join expression if the conditional is True; otherwise None
-    """
-    return expression if evaluator.eval_expression(condition) else None
-
-
-@macro()
-def where(
-    evaluator: MacroEvaluator,
-    condition: exp.Condition,
-    expression: exp.Where,
-) -> t.Optional[exp.Where]:
-    """Inserts WHERE expression when the condition is True
-
-    Example:
-        >>> from sqlglot import parse_one
-        >>> from sqlmesh.core.macros import MacroEvaluator
-        >>> sql = "select * from city @WHERE(True) population > 100 and country = 'Mexico'"
-        >>> MacroEvaluator().transform(parse_one(sql)).sql()
-        "SELECT * FROM city WHERE population > 100 AND country = 'Mexico'"
-
-    Args:
-        evaluator: MacroEvaluator that invoked the macro
-        condition: Condition expression
-        expression: Where expression
-    Returns:
-        Where expression if condition is True; otherwise None
-    """
-    return expression if evaluator.eval_expression(condition) else None
-
-
-@macro()
-def group_by(
-    evaluator: MacroEvaluator,
-    condition: exp.Condition,
-    expression: exp.Group,
-) -> t.Optional[exp.Group]:
-    """Inserts GROUP BY expression when the condition is True
-
-    Example:
-        >>> from sqlglot import parse_one
-        >>> from sqlmesh.core.macros import MacroEvaluator, group_by
-        >>> sql = "select * from city @GROUP_BY(True) country, population"
-        >>> MacroEvaluator().transform(parse_one(sql)).sql()
-        'SELECT * FROM city GROUP BY country, population'
-
-    Args:
-        evaluator: MacroEvaluator that invoked the macro
-        condition: Condition expression
-        expression: Group expression
-    Returns:
-        Group expression if the condition is True; otherwise None
-    """
-    return expression if evaluator.eval_expression(condition) else None
-
-
-@macro()
-def having(
-    evaluator: MacroEvaluator,
-    condition: exp.Condition,
-    expression: exp.Having,
-) -> t.Optional[exp.Having]:
-    """Inserts HAVING expression when the condition is True
-
-    Example:
-        >>> from sqlglot import parse_one
-        >>> from sqlmesh.core.macros import MacroEvaluator
-        >>> sql = "select * from city group by country @HAVING(True) population > 100 and country = 'Mexico'"
-        >>> MacroEvaluator().transform(parse_one(sql)).sql()
-        "SELECT * FROM city GROUP BY country HAVING population > 100 AND country = 'Mexico'"
-
-    Args:
-        evaluator: MacroEvaluator that invoked the macro
-        condition: Condition expression
-        expression: Having expression
-    Returns:
-        Having expression if the condition is True; otherwise None
-    """
-    return expression if evaluator.eval_expression(condition) else None
-
-
-@macro()
-def order_by(
-    evaluator: MacroEvaluator,
-    condition: exp.Condition,
-    expression: exp.Order,
-) -> t.Optional[exp.Order]:
-    """Inserts ORDER BY expression when the condition is True
```

Comment:
i don't like how we lose documentation, any way we can keep these docs? perhaps in _optional expression?

Reply:
I added the examples back to the `_optional_expression`. These run on doc tests and show usage for the various permutations. Is this what you meant?

Reply:
also if its not obvious in the diff BTW, `_optional_expression` already had docs, so we werent losing docs but dropping strings that are 90% repeated.

Reply:
exactly